### PR TITLE
feat(cmf/sagaRouter): add isExact to saga router

### PR DESCRIPTION
**Pull request by yyanwang**
**What is the problem this PR is trying to solve?**
If you defined a router saga on /resources and /resources/:id, and then when you access /resources/:id, saga of /resources will be triggered too.

should let user know if it's an exact match.
**What is the chosen solution to this problem?**
add `isExact` param to route saga generator, also add configuration of `restartOnRouteChange` to make route saga restarted when change to another matching route.
**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/cmf/__tests__/sagaRouter/router.test.js [hunk 1]
```diff
@@ -229,11 +268,53 @@ describe('sagaRouter route and route params', () => {
...
+						return {
+							pathname: '/resources',
+						};
+					}
+					return {
+						pathname: '/resources/action',
+					};
+				},
+			};
+		}
```
Comment (talend-glorieux) [inline packages/cmf/__tests__/sagaRouter/router.test.js]: Looks like that function is duplicated here https://github.com/Talend/ui/pull/1018/files#diff-aa898b17c94b680d677cafaa355f9b62R182. Could you extract it to define it only once?

**Comment by acateland** [comment]
find a missing pattern to be fixed, added a failling test.

File: packages/cmf/__tests__/sagaRouter/router.test.js [hunk 2]
```diff
@@ -165,12 +170,46 @@ describe('sagaRouter RouteChange', () => {
...
 		expect(anotherGen.next({ type: '@@router/LOCATION_CHANGE' }).value).toEqual(
-			spawn(alternateRoutes['/toCancelFirst'], {})
+			spawn(alternateRoutes['/toCancelFirst'], {}, true)
 		);
 		expect(anotherGen.next(mockTask).value).toEqual(take('@@router/LOCATION_CHANGE'));
 		const anotherExpectedCancelYield = cancel(mockTask);
 		expect(anotherGen.next({ type: '@@router/LOCATION_CHANGE' }).value).toEqual(anotherExpectedCancelYield);
 	});
+
+	it('saga will NOT be restarted when change to another matching location', () => {
```
[the remark this answers is not in the document]
Comment (yyanwang) [reply packages/cmf/__tests__/sagaRouter/router.test.js]: yes, you're right. will remove this one.

File: packages/cmf/src/sagaRouter/router.js [hunk 3]
```diff
@@ -146,27 +154,33 @@ function parseSagaState(routeFragment, sagas, currentLocation) {
 export default function* sagaRouter(history, routes) {
 	const sagas = {};
 	const routeFragments = Object.keys(routes);
-	while (true) {  // eslint-disable-line no-constant-condition
+	while (true) {
+		// eslint-disable-line no-constant-condition
```
Comment (jmfrancois) [inline packages/cmf/src/sagaRouter/router.js]: this doesn't work, bad linter ?

File: output/cmf.eslint.txt [hunk 4]
```diff
@@ -17,8 +17,9 @@ The react/require-extension rule is deprecated. Please use the import/extensions
   50:4  warning  Unexpected console statement  no-console
 
 /home/travis/build/Talend/ui/packages/cmf/src/sagaRouter/router.js
-  153:54  error  There should be no spaces inside this paren  space-in-parens
-  166:51  error  There should be no spaces inside this paren  space-in-parens
+  157:2   warning  Unexpected constant condition                no-constant-condition
```
Comment (jmfrancois) [inline output/cmf.eslint.txt]: see below

File: packages/cmf/__tests__/sagaRouter/router.test.js (hunk 1 above)
Comment (yyanwang) [reply packages/cmf/__tests__/sagaRouter/router.test.js]: I've modify this test case, now they test on different cases.

File: packages/cmf/src/sagaRouter/router.js (hunk 3 above)
Comment (yyanwang) [reply packages/cmf/src/sagaRouter/router.js]: yes, it's committed by CI's prettier :disappointed:

**Pull request merged**